sdk/communication/azure-communication-email/src/main/java/com/azure/communication/email/EmailClientBuilder.java

```diff
@@ -240,6 +266,18 @@ private AzureCommunicationServicesClientImpl buildInnerClient() {
         return client;
     }
 
+    private HttpPipelinePolicy createHttpPipelineAuthPolicy() {
+        if (this.tokenCredential != null) {
+            return new BearerTokenAuthenticationPolicy(
+                this.tokenCredential, "https://communication.azure.com//.default");
+        } else if (this.azureKeyCredential != null) {
+            return new HmacAuthenticationPolicy(this.azureKeyCredential);
+        } else {
+            throw logger.logExceptionAsError(
+                new IllegalArgumentException("Missing credential information while building a client."));
```

Comment:
An `IllegalStateException` is probably better here. We probably should also tell users how to remedy the problem (i.e. use one of the `credential()` methods to set a credential object). Any thoughts on this @srnagar?
```suggestion
                new IllegalStateException("Missing credential information while building a client."));
```